commerce-payment-web/src/main/resources/META-INF/resources/payment_method_action.jsp

```diff
@@ -36,37 +53,121 @@ CommercePaymentMethodGroupRel commercePaymentMethodGroupRel = (CommercePaymentMe
 			<portlet:param name="<%= Constants.CMD %>" value="<%= Constants.EDIT %>" />
 			<portlet:param name="redirect" value="<%= currentURL %>" />
 			<portlet:param name="commercePaymentMethodGroupRelId" value="<%= String.valueOf(commercePaymentMethodGroupRel.getCommercePaymentMethodGroupRelId()) %>" />
-			<portlet:param name="engineKey" value="<%= commercePaymentMethodGroupRel.getEngineKey() %>" />
+			<portlet:param name="engineKey" value="<%= engineKey %>" />
 		</portlet:actionURL>
 
 		<liferay-ui:icon
 			message="edit"
 			url="<%= editURL %>"
 		/>
 
-		<portlet:actionURL name="editCommercePaymentMethodGroupRel" var="setActiveURL">
-			<portlet:param name="<%= Constants.CMD %>" value="setActive" />
-			<portlet:param name="redirect" value="<%= currentURL %>" />
-			<portlet:param name="commercePaymentMethodGroupRelId" value="<%= String.valueOf(commercePaymentMethodGroupRel.getCommercePaymentMethodGroupRelId()) %>" />
-			<portlet:param name="active" value="<%= String.valueOf(!commercePaymentMethodGroupRel.isActive()) %>" />
-			<portlet:param name="engineKey" value="<%= commercePaymentMethodGroupRel.getEngineKey() %>" />
-		</portlet:actionURL>
-
 		<liferay-ui:icon
+			id='<%= "togglePaymentMethod-" + modalId %>'
 			message='<%= commercePaymentMethodGroupRel.isActive() ? "deactivate" : "activate" %>'
-			url="<%= setActiveURL %>"
+			url="<%= commercePaymentMethodGroupRel.isActive() ? StringPool.POUND : setActiveURL %>"
 		/>
 
 		<portlet:actionURL name="editCommercePaymentMethodGroupRel" var="editURL">
 			<portlet:param name="<%= Constants.CMD %>" value="<%= Constants.DELETE %>" />
 			<portlet:param name="redirect" value="<%= currentURL %>" />
 			<portlet:param name="commercePaymentMethodGroupRelId" value="<%= String.valueOf(commercePaymentMethodGroupRel.getCommercePaymentMethodGroupRelId()) %>" />
-			<portlet:param name="engineKey" value="<%= commercePaymentMethodGroupRel.getEngineKey() %>" />
+			<portlet:param name="engineKey" value="<%= engineKey %>" />
 		</portlet:actionURL>
 
 		<liferay-ui:icon
 			message="delete"
 			url="<%= editURL %>"
 		/>
 	</c:if>
-</liferay-ui:icon-menu>
\ No newline at end of file
+</liferay-ui:icon-menu>
+
+<c:choose>
+	<c:when test="<%= ((subscriptionEntryCount > 0) || isLastRecurringPaymentMethod) && commercePaymentMethodGroupRel.isActive() %>">
+		<style>
+			.warning-modal .modal-header {
+				background-color: #feefef;
+				border-color: #f48989;
+				color: #da1414;
+			}
+
+			.warning-modal p {
+				font-size: 16px;
+			}
```

Comment:
this seems..... overly specific. If there's any way we can add more classes to the `div` to hook into existing styling, rather than adding this custom styling just for this case (which might easily clash with somebody's theme down the road), that would be much better.